Sources/TuistGenerator/Generator/ProjectFileElements.swift

```diff
@@ -166,7 +166,7 @@ class ProjectFileElements {
     func generate(products: [String],
                   groups: ProjectGroups,
                   pbxproj: PBXProj) {
-        products.sorted().forEach { productName in
```

Comment:
What's the reason behind removing this `sorted()` and sorting the elements after they've been added to the Xcode project?

Reply:
I know it looks weird and I'd probably say the same seeing it. It's not that I removed the sorting from one place and put it somewhere else.
I think the sorting here is unnecessary. It's a bit tricky to explain without sharing the screen. Let's check out latest master (a8eb839705b4f66fa6dcd7fad2be840f38186f9e), btw sorry for that dummy explanation...

1. `ProjectFileElements.swift:72`: `generate(products: products.sorted(),` - sorts the products first time
2. When we go to that generate method, `ProjectFileElemenets.swift:169` -  `products.sorted().forEach { productName in`, products are sorted again (when the array list is already sorted as there is only one caller, line 72).
3. We would remove one of the `sorted` called but that won't help, the issue is still there. The products array is sorted - but only initially

in `ProjectFileElements.swift`, after 
```
        /// Products
        generate(products: products.sorted(),
                 groups: groups,
                 pbxproj: pbxproj)
```

we got

```
        /// Dependencies
        try generate(dependencies: dependencies,
                     path: project.path,
                     groups: groups,
                     pbxproj: pbxproj,
                     sourceRootPath: sourceRootPath,
                     filesGroup: project.filesGroup)
```

and inside that `generate methods`, line 224, we call
```
             groups.products.children.append(fileReference)
```

that simply breaks the order.

At least, that's my understanding of the cause.

Reply:
Got you! Then I'd suggest a simpler approach. The current implementation is:

1. We add the products (sorted) for each of the project targets.
2. When we generate the dependencies, we add their products to the existing *(and sorted)* list of products. 
3. As a result, we get a non-sorted list of products in the project.

Instead of doing that we can:

1. Get the list of products, that includes the products of the targets and their dependencies.
2. Sort them.
3. Add them to the products group.

How does that sound?

Reply:
Thanks for the advice, i wanted to do that from the beginning but as 
```swift
    func generate(dependencies: Set<GraphNode>,
                  path: AbsolutePath,
                  groups: ProjectGroups,
                  pbxproj: PBXProj,
                  sourceRootPath: AbsolutePath,
                  filesGroup: ProjectGroup)
```

can also modify other groups, I thought it would require bigger amount of changes. 

Had a second look and found quite clean way of doing it, I'll push it soon - thanks again 👍